adage/analyze/api.py

```diff
@@ -427,7 +427,14 @@ def apply_filters(self, request, applicable_filters):
                 raise BadRequest("Invalid gene IDs: %s" % genes)
             # "in" operator in Django supports both list and set.
             qset = Q(gene1__in=ids) | Q(gene2__in=ids)
-            object_list = object_list.filter(qset).distinct()
+            direct_edges = Edge.objects.filter(qset).distinct()
+            related_genes = set()
+            for e in direct_edges:
+                related_genes.add(e.gene1)
+                related_genes.add(e.gene2)
+            object_list = object_list.filter(
+                gene1__in=related_genes, gene2__in=related_genes
+            ).distinct()
```

Comment:
@mhuyck: Your proposal is actually equivalent to my current implementation. When there are multiple conditions inside `filter`, these conditions are AND-ed together. (What you suggested was my original version, but I changed it to the current version because I feel that using `filter` will be easier to understand.